> -1
> 
> Tested at: [fdfaf1b](https://github.com/cms-sw/cmssw/commit/fdfaf1bb756ab12ea020146dd9d112a67928f25d)
> 
> CMSSW: CMSSW_11_2_X_2020-06-30-2300
> SCRAM_ARCH: slc7_amd64_gcc820
> 
> The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic:
> 
>     * #30479 @silviodonato: Turn on L1 track trigger in Phase2C8 and later
> 
>     * #30363 @cms-l1t-offline: New Package - Phase2 L1TTrackMatch
> 
>     * #30480 @silviodonato: Move L1TrackTrigger step before L1 step
> 
> 
> You can see more details here:
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5f96b/7562/git-log-recent-commits
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5f96b/7562/git-merge-result
> 
> You can see the results of the tests here:
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5f96b/7562/summary.html
> 
> I found follow errors while testing this PR
> 
> **Failed tests: ClangBuild**
> 
>     * **Clang**:
> 
> 
> I found compilation warning while trying to compile with clang. Command used:
> 
> ```
> USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'
> ```
> 
> See details on the summary page.
> 
> The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic:
> 
>     * #30479 @silviodonato: Turn on L1 track trigger in Phase2C8 and later
> 
>     * #30363 @cms-l1t-offline: New Package - Phase2 L1TTrackMatch
> 
>     * #30480 @silviodonato: Move L1TrackTrigger step before L1 step
> 
> 
> You can see more details here:
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5f96b/7562/git-log-recent-commits
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5f96b/7562/git-merge-result

@smuzaffar @makortel @mrodozov the only warning I see is 
```
L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc
/cvmfs/cms-ib.cern.ch/nweek-02635/slc7_amd64_gcc820/external/tbb/2020_U2/include/tbb/task.h:21:9: warning: In file included from TBB Warning: tbb/task.h is deprecated. For details, please see Deprecated Features appendix in the TBB reference manual. [-W#pragma-messages]/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-30-2300/src/L1Trigger/L1TTrackMatch/plugins/L1TkFastVertexProducer.cc
```
I remember this error should be ignore. Moreover `L1Trigger/L1TTrackMatch` was added by another PR and the test were successful.

--- a/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
+++ b/Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py
@@ -672,6 +672,11 @@ class UpgradeWorkflow_TestOldDigi(UpgradeWorkflow):
             # handle separate PU input
             stepNamePU = step + 'PU' + self.suffix
             stepDict[stepNamePU][k] = merge([{'--filein': 'das:/RelValTTbar_14TeV/CMSSW_11_0_0_pre13-PU25ns_110X_mcRun4_realistic_v2_2026D49PU200-v2/GEN-SIM-DIGI-RAW'},stepDict[stepName][k]])
+            ## Use re-emulate the full L1 trigger when running on 11_0 DIGI. Ie. Replace L1Reco with L1TrackTrigger,L1. 
+            stepDict[stepName][k] = merge([{'-s': stepDict[stepName][k]['-s'].replace("L1Reco","L1TrackTrigger,L1")}, stepDict[stepName][k]])
+            stepDict[stepNamePU][k] = merge([{'-s': stepDict[stepNamePU][k]['-s'].replace("L1Reco","L1TrackTrigger,L1")}, stepDict[stepNamePU][k]])
+            stepDict[stepName][k] = merge([{'--customise_unsch': "L1Trigger/Configuration/customisePhase2TTNoMC.customisePhase2TTNoMC"}, stepDict[stepName][k]])
+            stepDict[stepNamePU][k] = merge([{'--customise_unsch': "L1Trigger/Configuration/customisePhase2TTNoMC.customisePhase2TTNoMC"}, stepDict[stepNamePU][k]])
         elif 'GenSim' in step or 'Digi' in step:
             # remove step
             stepDict[stepName][k] = None

--- a/Configuration/StandardSequences/python/L1TrackTrigger_cff.py
+++ b/Configuration/StandardSequences/python/L1TrackTrigger_cff.py
@@ -14,4 +14,4 @@ _tttracks_l1tracktrigger = cms.Sequence(_tttracks_l1tracktrigger + L1PromptExten
 from Configuration.Eras.Modifier_phase2_trackerV14_cff import phase2_trackerV14
 phase2_trackerV14.toReplaceWith( L1TrackTrigger, _tttracks_l1tracktrigger )
 
-TTStubAlgorithm_official_Phase2TrackerDigi_.zMatchingPS = cms.bool(True)
+TTStubAlgorithm_official_Phase2TrackerDigi_.zMatchingPS = True

--- a/L1Trigger/Configuration/python/customisePhase2TTNoMC.py
+++ b/L1Trigger/Configuration/python/customisePhase2TTNoMC.py
@@ -1,0 +1,4 @@
+def customisePhase2TTNoMC(process):
+    process.L1TrackTrigger.replace(process.L1PromptExtendedHybridTracksWithAssociators, process.L1PromptExtendedHybridTracks)
+    process.L1TrackTrigger.remove(process.TrackTriggerAssociatorClustersStubs)
+    return process

